lsm_tree.h

```diff
@@ -0,0 +1,44 @@
+#ifndef LSM_TREE_H
+#define LSM_TREE_H
+
+#include "memtable.h"
+
+#define NUM_OPTIONS 6          // number of actions LSM tree can do
+#define STR_BUF 5              // leave plenty of room for options
+#define MAX_LEN_KEYS 10        // max size of key (10 digits)
+#define MAX_LEN_DATA 50        // max length of data for value in database
+#define MAX_KEYS_IN_TREE 3     // max # keys held in tree before flush to segment
+#define MAX_SEGMENTS 2         // max # full segments before compaction
+#define FILENAME_SIZE 20       // file name size
+#define MAX_LINE_SIZE 60       // max number of characters in a single line of a segment file
+
+enum available_actions {
+	ADD = 1, SEARCH = 2, DELETE = 3, SAVE = 4, PRINT_MEMTABLE = 5, EXIT = 6
+};
+
+typedef struct user_submission {
+	enum available_actions action;
+	int key;
```

Comment:
Just curious why is ```key``` an ```int```? Personal preference would be a ```char *```

Reply:
Yea, that's a good point. It probably should be a char*. Using a key is easy for making comparisons and order which was why I decided to go with that first. Will change in a next iteration.